src/test/common/channel-control.ts

```diff
@@ -19,9 +19,9 @@ export interface ChannelControl<X, Y> {
   
     // listening
     initCompleteListener(testId: string): Promise<Y>
-    setupAndValidateListener1(channel: X, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
-    setupAndValidateListener2(channel: X, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
-    setupContextChecker(channel: X, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): Promise<void>;
+    setupAndValidateListener1(channel: X | null, expectedContextType: string | null, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
```

Comment:
I think we ought to separate the expectedContextType and the type of the listener added - particularly as I think the expected context type always has to be set.

```suggestion
    setupAndValidateListener1 = (channel: X | null, listenerContextType: string | null, expectedContextType: string, errorMessage: string, onComplete: (ctx: Y) => void): void | Promise<void>;
```

Reply:
Not true - I found more than one test where a null listener should be added and a specified type received.